tests/integration/suite/daprd/resiliency/apps/defaultcircuitbreaker.go

```diff
@@ -0,0 +1,197 @@
+}
+
+type defaultcircuitbreaker struct {
+	daprdClient *daprd.Daprd
+	daprdServer *daprd.Daprd
+	callCount1  *atomic.Int32
+	callCount2  *atomic.Int32
+}
+
+func (d *defaultcircuitbreaker) Setup(t *testing.T) []framework.Option {
+	d.callCount1 = &atomic.Int32{}
+	d.callCount2 = &atomic.Int32{}
+
+	handler := http.NewServeMux()
+	handler.HandleFunc("/circuitbreaker_ok", func(w http.ResponseWriter, r *http.Request) {
+		d.callCount1.Add(1)
+		w.WriteHeader(http.StatusOK)
+	})
+	handler.HandleFunc("/circuitbreaker_fail", func(w http.ResponseWriter, r *http.Request) {
+		d.callCount2.Add(1)
+		if d.callCount2.Load() <= 3 {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+	})
+
+	srv := prochttp.New(t, prochttp.WithHandler(handler))
+
+	resiliency := `
+apiVersion: dapr.io/v1alpha1
+kind: Resiliency
+metadata:
+  name: myresiliency
+spec:
+  policies:
+    circuitBreakers:
+      DefaultCircuitBreaker:
+        maxRequests: 1
+        interval: 0
+        timeout: 5s
+        trip: consecutiveFailures >= 3
+  targets:
+    apps:
+      server:
+        circuitBreaker: DefaultCircuitBreaker
+`
+
+	d.daprdClient = daprd.New(t,
+		daprd.WithAppPort(srv.Port()),
+		daprd.WithResourceFiles(resiliency),
+		daprd.WithAppID("client"),
+		daprd.WithLogLevel("debug"),
+	)
+	d.daprdServer = daprd.New(t,
+		daprd.WithAppPort(srv.Port()),
+		daprd.WithAppID("server"),
+	)
+	// log.Print(d.daprdClient.Metrics(t, context.Background()))
+
+	return []framework.Option{
+		framework.WithProcesses(srv, d.daprdClient, d.daprdServer),
+	}
+}
+
+func (d *defaultcircuitbreaker) Run(t *testing.T, ctx context.Context) {
+	d.daprdClient.WaitUntilRunning(t, ctx)
+	d.daprdServer.WaitUntilRunning(t, ctx)
+
+	t.Run("circuit breaker does not open after consecutive successes", func(t *testing.T) {
+		reqURL := fmt.Sprintf("http://localhost:%d/v1.0/invoke/%s/method/circuitbreaker_ok", d.daprdClient.HTTPPort(), d.daprdServer.AppID())
+
+		// First 3 calls should pass
+		for i := 1; i <= 3; i++ {
+			req, err := http.NewRequestWithContext(ctx, http.MethodGet, reqURL, nil)
+			require.NoError(t, err)
+			resp, err := client.HTTP(t).Do(req)
+			require.NoError(t, err)
+			assert.Equal(t, http.StatusOK, resp.StatusCode)
+			require.NoError(t, resp.Body.Close())
+		}
+
+		// 4th call should not trigger circuit breaker
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, reqURL, nil)
+		require.NoError(t, err)
+		resp, err := client.HTTP(t).Do(req)
+		require.NoError(t, err)
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+		require.NoError(t, err)
+		require.NoError(t, resp.Body.Close())
+
+		// assert cb execution,activation,and current state counts
+		mtc := d.daprdClient.Metrics(t, context.Background())
+		assert.InDelta(t, float64(4), mtc["dapr_resiliency_count|app_id:client|flow_direction:outbound|name:myresiliency|namespace:|policy:circuitbreaker|status:closed|target:app_server"], 0)
```

Comment:
You can disable this lint check that requires `InDelta` and revert to the integer or equals assert.
See example in `component-contrib` repo https://github.com/dapr/components-contrib/blob/release-1.14/.golangci.yml#L266

Reply:
Does that disable float compare for the entire repo? I see some usage of `//nolint:testifylint` on a per file, possibly per line basis, but I'm fine using the InDelta as I switched to.

Reply:
> Does that disable float compare for the entire repo?

Yes. It is indeed might be excessive. That's NIT anyway.